numpy/_core/tests/test_memmap.py

```diff
@@ -207,10 +214,13 @@ def test_no_shape(self):
     def test_empty_array(self):
         # gh-12653
         with pytest.raises(ValueError, match='empty file'):
-            memmap(self.tmpfp, shape=(0,4), mode='w+')
+            memmap(self.tmpfp, shape=(0,4), mode='r')
 
-        self.tmpfp.write(b'\0')
+        # gh-27723
+        # empty memmap works with mode in ('w+','r+')
+        memmap(self.tmpfp, shape=(0,4), mode='w+')
 
+        self.tmpfp.write(b'\0')
         # ok now the file is not empty
         memmap(self.tmpfp, shape=(0,4), mode='w+')
```

Comment:
```suggestion
        memmap(self.tmpfp, shape=(0, 4), mode='w+')
```
Can fix if it if you don't mind. Also happy to ignore it though :).